test/plugin/test_out_secondary_file.rb

```diff
@@ -0,0 +1,308 @@
+require_relative '../helper'
+require 'time'
+require 'fileutils'
+require 'fluent/test'
+require 'fluent/event'
+require 'fluent/plugin/buffer'
+require 'fluent/plugin/out_secondary_file'
+require 'fluent/plugin/buffer/memory_chunk'
+
+class FileOutputSecondaryTest < Test::Unit::TestCase
+  def setup
+    Fluent::Test.setup
+    FileUtils.rm_rf(TMP_DIR)
+    FileUtils.mkdir_p(TMP_DIR)
+  end
+
+  TMP_DIR = File.expand_path(File.dirname(__FILE__) + "/../tmp/out_file#{ENV['TEST_ENV_NUMBER']}")
+
+  CONFIG = %[
+    path #{TMP_DIR}/out_file_test
+    compress gz
+  ]
+
+  class DummyOutput < Fluent::Plugin::Output
+    def write(chunk); end
+  end
+
+  def create_primary(buffer_cofig = config_element('buffer'))
+    DummyOutput.new.configure(config_element('ROOT','',{}, [buffer_cofig]))
+  end
+
+  def create_driver(conf = CONFIG, primary = create_primary)
+    c = Fluent::Test::OutputTestDriver.new(Fluent::Plugin::SecondaryFileOutput)
```

Comment:
`Fluent::Test::OutputTestDriver` is for plugins on v0.12 (old) APIs.
Use `fluent/test/driver/output` instead.